demosplan/DemosPlanCoreBundle/Security/Authentication/Provider/SecurityUserProvider.php

```diff
@@ -63,7 +64,7 @@ public function getSecurityUser(string $identifier): SecurityUser
         return new SecurityUser($this->loadUserByLogin($identifier));
     }
 
-    public function upgradePassword(UserInterface $user, string $newHashedPassword): void
+    public function upgradePassword(UserInterface|PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void
```

Comment:
It is similar to other case:
- what about using only PasswordAuthenticatedUserInterface. UserInterface is already extending from PasswordAuthenticatedUserInterface
- Maybe remove the doc comment?

Reply:
same as above as it is the same implementation